### PR TITLE
course.kcl visualization

### PR DESCRIPTION
**Pull request by riidefi**
Three modes:
1. Disabled
2. "Overlay" - Draw on top of the existing game visuals
3. "Replace" - Hide the game visuals (course_model.brres) and only render visual course.kcl

Footage: https://www.youtube.com/watch?v=oaOMoB3t5UE

### CODE REVIEW

File: symbols.txt [hunk 1]
```diff
@@ -176,6 +178,7 @@
 0x8016e5a4 GXSetNumTexGens
 0x8016e9fc GXSetDrawSync
 0x8016eab0 GXDrawDone
+0x8016e37c GXSetTexCoordGen2
```
Comment (stblr) [inline symbols.txt:181]: Order

File: port.py [hunk 2]
```diff
@@ -529,6 +530,7 @@ def port(self, address):
         Chunk(0x809c2be8, 0x809c2bec, 0x809c1c48),
         Chunk(0x809c2ef0, 0x809c2ef4, 0x809c1f50),
         Chunk(0x809c2f38, 0x809c2f40, 0x809c1f98),
+        Chunk(0x809c2f44, 0x908c2f48, 0x809b1584),
```
Comment (stblr) [inline port.py:533]: C/P from K?

File: assets/message/MenuSP_U.bmg.json5 [hunk 3]
```diff
@@ -1230,6 +1230,34 @@
...
+        "string": "Show on top of course_model.brres",
+    },
+    "10353": {
+        "font": "regular",
+        "string": "Replace course_model.brres",
+    },
+    "10354": {
+        "font": "regular",
+        "string": "Hide course_model.brres",
+    },
```
Comment (stblr) [inline assets/message/MenuSP_U.bmg.json5:1260]: Outdated?

File: payload/sp/3d/Kcl.cc [hunk 4]
```diff
@@ -0,0 +1,390 @@
...
+        clr |= 0xff;
+        // U16 for UVs for alignment (*2 = 4 bytes)
+        vertices[i * 3] = {verts[0], fnrm, clr, 0, 0};
+        vertices[i * 3 + 1] = {verts[1], fnrm, clr, 0, 1};
+        vertices[i * 3 + 2] = {verts[2], fnrm, clr, 1, 1};
+    }
+
+    if (!trussBound) {
+        trussBound = true;
+        YAZDecoder::Decode(TRUSS_SZS, sizeof(TRUSS_SZS), TRUSS_TPL, sizeof(TRUSS_TPL));
```
Comment (stblr) [inline payload/sp/3d/Kcl.cc:294]: What's the benefit of compressing this if the memory is statically allocated?

**Review by stblr — CHANGES_REQUESTED** [review]

File: payload/sp/3d/Kcl.cc (hunk 4 above)
Comment (riidefi) [reply payload/sp/3d/Kcl.cc:294]: Lines of code in the diff

File: assets/message/MenuSP_U.bmg.json5 (hunk 3 above)
Comment (riidefi) [reply assets/message/MenuSP_U.bmg.json5:1260]: Should be fine

File: port.py (hunk 2 above)
Comment (riidefi) [reply port.py:533]: Good catch

**Review by stblr — APPROVED** [review]

**Pull request merged**